// @daimler/ftk-core/lib/router/lib/RouterService.js
"use strict";
// SPDX-License-Identifier: MIT
// Copyright (c) 2020 Daimler TSS GmbH
Object.defineProperty(exports, "__esModule", { value: true });
var RouterService = /** @class */ (function () {
    function RouterService(router, urlFormatStrategy) {
        this.urlFormatStrategy = urlFormatStrategy;
        this.router = router;
    }
    RouterService.prototype.getRoute = function () {
        var _a = this.router.getActiveRoute(), name = _a.name, url = _a.url, parameter = _a.parameter, query = _a.query;
        return { name: name, url: url, parameter: parameter, query: query };
    };
    RouterService.prototype.linkTo = function (name, parameter, query) {
        if (parameter === void 0) { parameter = {}; }
        if (query === void 0) { query = {}; }
        return this.urlFormatStrategy.format(this.router.linkTo(name, parameter, query));
    };
    RouterService.prototype.linkToHome = function () {
        return this.urlFormatStrategy.format('/');
    };
    RouterService.prototype.navigate = function (url) {
        window.location.href = url;
    };
    RouterService.prototype.navigateToHome = function () {
        this.navigate(this.urlFormatStrategy.format('/'));
    };
    RouterService.$$injectionKey = 'RouterService';
    return RouterService;
}());
exports.RouterService = RouterService;
