// @daimler/ftk-core/lib/util/url/UrlUtil.js
"use strict";
// SPDX-License-Identifier: MIT
// Copyright (c) 2020 Daimler TSS GmbH
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @class Url
 * @implements IUrl
 */
var Url = /** @class */ (function () {
    /**
     * @constructor
     * @param {IUrlStrategy} strategy
     */
    function Url(strategy) {
        this.strategy = strategy;
    }
    /**
     * @param url
     * @returns {string[]}
     */
    Url.prototype.extractPath = function (url) {
        return this.strategy.extractPath(url);
    };
    /**
     * @param segments
     * @returns {string}
     */
    Url.prototype.buildPathFromSegments = function (segments) {
        return this.strategy.buildPathFromSegments(segments);
    };
    /**
     * @param url
     * @returns {string}
     */
    Url.prototype.extractHashbang = function (url) {
        return this.strategy.extractHashbang(url);
    };
    /**
     * @param hash
     * @returns {string}
     */
    Url.prototype.buildHashbang = function (hash) {
        return this.strategy.buildHashbang(hash);
    };
    /**
     * @param url
     * @returns {any}
     */
    Url.prototype.extractQuery = function (url) {
        return this.strategy.extractQuery(url);
    };
    /**
     * @returns {string}
     */
    Url.prototype.buildQueryFromParameters = function (params) {
        if (params === void 0) { params = {}; }
        return this.strategy.buildQueryFromParameters(params);
    };
    return Url;
}());
exports.default = Url;
