// @daimler/ftk-core/lib/util/url/strategy/DefaultUrlStrategy.js
"use strict";
// SPDX-License-Identifier: MIT
// Copyright (c) 2020 Daimler TSS GmbH
Object.defineProperty(exports, "__esModule", { value: true });
var UrlHelper_1 = require("../UrlHelper");
/**
 * @class Default
 */
var DefaultUrlStrategy = /** @class */ (function () {
    function DefaultUrlStrategy() {
    }
    /**
     * @param url
     * @returns {string[]}
     */
    DefaultUrlStrategy.prototype.extractPath = function (url) {
        return new UrlHelper_1.default(url).getSegments().filter(function (segment) {
            return segment.length > 0;
        });
    };
    /**
     * @param segments
     * @returns {string}
     */
    DefaultUrlStrategy.prototype.buildPathFromSegments = function (segments) {
        var buffer = new UrlHelper_1.default().setSegmentsToString(segments);
        if (buffer.substr(0, 1) !== '/') {
            buffer = '/' + buffer;
        }
        return decodeURIComponent(buffer);
    };
    /**
     * @param url
     * @returns {string}
     */
    DefaultUrlStrategy.prototype.extractHashbang = function (url) {
        return new UrlHelper_1.default(url).extractHash();
    };
    /**
     * @param hash
     * @returns {string}
     */
    DefaultUrlStrategy.prototype.buildHashbang = function (hash) {
        return '#!' + hash;
    };
    /**
     * @param url
     * @returns {IUrlQueryMap}
     */
    DefaultUrlStrategy.prototype.extractQuery = function (url) {
        return new UrlHelper_1.default(url).getQueryMap();
    };
    /**
     * @returns {string}
     */
    DefaultUrlStrategy.prototype.buildQueryFromParameters = function (params) {
        if (params === void 0) { params = {}; }
        return new UrlHelper_1.default().setQueryString(params);
    };
    return DefaultUrlStrategy;
}());
exports.default = DefaultUrlStrategy;
