// @daimler/ftk-core/lib/util/loader/ScriptLoader.js
"use strict";
// SPDX-License-Identifier: MIT
// Copyright (c) 2020 Daimler TSS GmbH
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * A helper class to load additional scripts in runtime.
 *
 * @class ScriptLoader
 * @param url: string
 */
var ScriptLoader = /** @class */ (function () {
    function ScriptLoader(url) {
        var _this = this;
        /**
         * Create a new script tag, wet it in head and load the script with a callback,
         * which will be called after loading the script.
         *
         * @name loadScript
         * @return Promise<any>
         */
        this.loadScript = function () {
            return new Promise(function (resolve, reject) {
                var scriptTag = _this.doc.createElement('script');
                var onResolve = function () {
                    resolve();
                };
                scriptTag.onload = onResolve;
                scriptTag.onerror = function (event, source, lineno, colno, error) {
                    reject({ event: event, error: error });
                };
                // IE specific event ('onreadystatechange')
                scriptTag.onreadystatechange = function () {
                    if (scriptTag.readyState && scriptTag.readyState === 'complete') {
                        onResolve();
                    }
                };
                scriptTag.src = _this.url;
                _this.head.insertBefore(scriptTag, _this.head.lastChild);
            });
        };
        this.url = url;
        this.doc = document;
        this.head = this.doc.getElementsByTagName('head')[0];
    }
    return ScriptLoader;
}());
exports.ScriptLoader = ScriptLoader;
