// @daimler/ftk-core/lib/index.js
"use strict";
// SPDX-License-Identifier: MIT
// Copyright (c) 2020 Daimler TSS GmbH
Object.defineProperty(exports, "__esModule", { value: true });
// core
var kernel_1 = require("./core/kernel");
exports.kernel = kernel_1.default;
var App_1 = require("./core/App");
exports.App = App_1.App;
var serviceIds_1 = require("./core/serviceIds");
exports.serviceIds = serviceIds_1.default;
var ConfigService_1 = require("./config/lib/ConfigService");
exports.ConfigService = ConfigService_1.ConfigService;
var Provider_1 = require("./config/component/Provider");
exports.ConfigProvider = Provider_1.ConfigProvider;
// ErrorHandler
var ErrorHandler_1 = require("./errorHandler/lib/ErrorHandler");
exports.ErrorHandler = ErrorHandler_1.ErrorHandler;
var inversify_1 = require("inversify");
exports.DiContainer = inversify_1.Container;
// i18n
var I18nService_1 = require("./i18n/lib/I18nService");
exports.I18nService = I18nService_1.I18nService;
var composeTranslations_1 = require("./i18n/hoc/composeTranslations");
exports.composeTranslations = composeTranslations_1.default;
var TranslationProvider_1 = require("./i18n/component/TranslationProvider");
exports.TranslationProvider = TranslationProvider_1.TranslationProvider;
// Router
var Route_1 = require("./router/component/Route");
exports.Route = Route_1.default;
var RouterProvider_1 = require("./router/component/RouterProvider");
exports.RouterProvider = RouterProvider_1.default;
var RouterService_1 = require("./router/lib/RouterService");
exports.RouterService = RouterService_1.RouterService;
// DI
var BindToDi_1 = require("./di/component/BindToDi");
exports.BindToDi = BindToDi_1.default;
var ChildContainer_1 = require("./di/component/ChildContainer");
exports.ChildContainer = ChildContainer_1.default;
var withInject_1 = require("./di/hoc/withInject");
exports.withInject = withInject_1.default;
var inject_1 = require("./di/hoc/inject");
exports.inject = inject_1.default;
// hotloading
var SwidgetLoader_1 = require("./componentLoader/component/SwidgetLoader");
exports.SwidgetLoader = SwidgetLoader_1.default;
var InterconnectionService_1 = require("./interconnection/InterconnectionService");
exports.InterconnectionService = InterconnectionService_1.InterconnectionService;
var singleton_1 = require("./di/hoc/singleton");
exports.singleton = singleton_1.default;
// Hooks bridge
var Hooks_1 = require("./util/Hooks");
exports.useFromDi = Hooks_1.useFromDi;
exports.useConfig = Hooks_1.useConfig;
exports.useDi = Hooks_1.useDi;
exports.useErrorHandler = Hooks_1.useErrorHandler;
exports.useInterconnection = Hooks_1.useInterconnection;
exports.useRouter = Hooks_1.useRouter;
exports.useI18n = Hooks_1.useI18n;
